src/Controller/UserController.php

```diff
@@ -57,4 +52,91 @@ public function register(
             'registerForm' => $registerForm->createView()
         ]);
     }
+
+    /**
+     * @Route("/", name="user_index", methods={"GET"})
+     */
+    public function index(UserRepository $userRepository): Response
+    {
+        return $this->render('user/index.html.twig', [
+            'users' => $userRepository->findAll(),
+        ]);
+    }
+
+    /**
+     * @Route("/new", name="user_new", methods={"GET","POST"})
+     */
+    public function new(Request $request): Response
+    {
+        $user = new User();
+        $form = $this->createForm(UserType::class, $user);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->persist($user);
+            $entityManager->flush();
+
+            return $this->redirectToRoute('user_index');
+        }
+
+        return $this->render('user/new.html.twig', [
+            'user' => $user,
+            'form' => $form->createView(),
+        ]);
+    }
+
+    /**
+     * @Route("/{id}", name="user_show", methods={"GET"})
+     */
+    public function show(User $user): Response
+    {
+        return $this->render('user/show.html.twig', [
+            'user' => $user,
+        ]);
+    }
+
+    /**
+     * @Route("/{username}/edit", name="user_edit", methods={"GET","POST"})
+     */
+    public function edit(Request $request, User $user, UserPasswordEncoderInterface $encoder): Response
+    {
+        if($user->getEmail() != $this->getUser()->getEmail()){
+            return $this->redirectToRoute('app_home');
+        }
```

Comment:
La condition sert à vérifier si l'utilisateur essaie de modifier le profil de quelqu'un d'autre ?
Dans ce cas, c'est peut-être plus indiqué d'enlever {username} du path, et à chaque fois que tu dois référer au user, tu fais un $this->getUser(), à voir.